### ISSUE TITLE
Register error: Undefined variable: userEntity

### ISSUE THREAD

**Issue by ayman-alkom** [burzum/cakephp-user-tools#23]
the line `518` in `/src/Controller/Component/UserToolComponent.php` will not be executed in the case I submitted unsuccessful request because it will return from the line `515`, this cause error with `userData` is undefined.

**Comment by burzum** [comment]
Could you please properly report how to reproduce the issue?

**Comment by ayman-alkom** [comment]
I resolved the problem as the following:

```PHP
	public function register($options = []) {
		$options = Hash::merge($this->_config['registration'], $options);
		if ($options['enabled'] === false) {
			throw new NotFoundException();
		}
		$entity = $this->UserTable->newEntity();

		$return = false;                 // make return as a variable

		// Make the field accessible in the case the default entity class is used.
		$entity->accessible('confirm_password', true);
		if ($this->request->is('post')) {
			$entity = $this->UserTable->patchEntity($entity, $this->request->data());
			if ($this->UserTable->register($entity)) {
				$this->handleFlashAndRedirect('success', $options);
				if ($options['setEntity'] === true) {
					$this->_controller->set('usersEntity', $entity);
				}

				$return = true;    //dont return here, just set the value to true

			} else {
				$this->handleFlashAndRedirect('error', $options);
				if ($options['setEntity'] === true) {
					$this->_controller->set('usersEntity', $entity);
				}
			}
		}
        //this line will executed in all cases, no return before.
		if ($options['setEntity'] === true) {
			$this->_controller->set('userEntity', $entity);
			// BC
			$this->_controller->set('usersEntity', $entity);
		}

        //the return comes here
        return $return;
	}
```
I hope the issue is clear now, and sorry if my English is not very well :)

**Comment by burzum** [comment]
Can you please show the change as diff or create a pull request?

**Comment by burzum** [comment]
Fixed in the develop branch.

**Comment by ayman-alkom** [comment]
thanks

**Comment by ayman-alkom** [comment]
Sorry for my out of box question, but how can I use the develop branch?
I tried several times but didn't succeed

**Comment by burzum** [comment]
I urgently recommend you to read the composer manual. https://getcomposer.org/doc/

Composer is a common tool these days and Cake and almost all plugins use it.

```sh
composer require burzum/cakephp-user-tools dev-develop
```

But I've merged it into master as well and tagged 1.0.1 too.
